.github/workflows/test-gpu.yml

```diff
@@ -57,8 +57,11 @@ jobs:
           environment-file: ci/gpu_ci.yml
           post-cleanup: "all"
 
+      - name: Install cupy
+        run: uv pip install cupy-cuda12x numba pytest pytest-cov pytest-xdist
+
       - name: Install AnnData
-        run: uv pip install -e .[dev,test,gpu]
+        run: uv pip install -e .[dev,test]
```

Comment:
1. Why do we need these `pytest` deps if they come from `test` (Except for `xdist`)?
2. Why are we using `cupy` as the dep in `gpu` instead of a pre-built wheel?  
3. Since it seems like we need `numba` why not make a `test-gpu` section in the `pypoject.toml`?

Reply:
also please never mutate an environment after it has been created. that means

1. only use one command to create the environment, then use it.

   you can use `-e` in the same call as specifying other dependencies!

3. no `pip: ...` entry in an environment.yaml that’s installed by conda, as that translates to `conda install ... && pip install ...`. If you need to install conda packages and pip packages, use something that supports that, e.g. conda-lock or pixi